src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,96 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { postComment } from '../api/data';
+
+type Props = {
+  postId: number
+  getNewComments: (postId: number) => Promise<void>;
+  commentError: (param: boolean) => void
+};
+
+export const NewCommentForm: React.FC<Props>
+= React.memo(({ postId, getNewComments, commentError }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [comment, setComment] = useState('');
+
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isCommentError, setIsCommentError] = useState(false);
+  const [isCommentLoading, setIsCommentLoading] = useState(false);
+
+  const handleInput = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    callback: (param: string) => void,
+  ) => {
+    const { value } = event.currentTarget;
+
+    callback(value);
+  };
+
+  const handleClear = () => {
+    if (isEmailError) {
+      setIsEmailError(false);
+    }
+
+    if (isCommentError) {
+      setIsCommentError(false);
+    }
+
+    if (isNameError) {
+      setIsNameError(false);
+    }
+
+    if (name) {
+      setName('');
+    }
+
+    if (email) {
+      setEmail('');
+    }
+
+    if (comment) {
+      setComment('');
+    }
+  };
+
+  const postCommentOnServer = async () => {
+    if (!name) {
+      setIsNameError(true);
+    }
+
+    if (!email) {
+      setIsEmailError(true);
+    }
+
+    if (!comment) {
+      setIsCommentError(true);
+    }
+
+    if (name && email && comment) {
+      setIsCommentLoading(true);
+
+      postComment(
+        postId,
+        name.trim(),
+        email.trim(),
+        comment.trim(),
+      )
+        .then(() => getNewComments(postId))
+        .catch(() => commentError(true))
+        .finally(() => {
+          setComment('');
+          setIsCommentLoading(false);
+        });
+    }
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form
+      data-cy="NewCommentForm"
+      onSubmit={(event) => event.preventDefault()}
```

Comment:
Consider moving all logic for submitting the form to onSubmit event handler instead of onClick event for the submit button.

Reply:
Done